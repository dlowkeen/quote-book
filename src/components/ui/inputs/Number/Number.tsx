import React from 'react';

import * as styles from '../inputs.css';

interface INumberProps {
  max: string;
  label: string;
  value: string;
  onChange: (e: any) => void;
}

const Number: React.FunctionComponent<INumberProps> = ({
  label,
  value,
  onChange,
  max,
}) => (
  <div className={`${styles.formElement} ${styles.formElementText}`}>
    <label>{label}</label>
    <input type='number' value={value} onChange={onChange} max={max} min='0' />
  </div>
);

export default Number;
