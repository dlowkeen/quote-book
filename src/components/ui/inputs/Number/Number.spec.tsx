import React from 'react';
import renderer from 'react-test-renderer';

import Number from './Number';

describe('The Number component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <Number max='10' value='3' label='label' onChange={() => 'changed!'} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
