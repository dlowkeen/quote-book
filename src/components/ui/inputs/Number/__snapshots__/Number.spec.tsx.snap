// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Number component renders as expected 1`] = `
<div
  className="inputs__form-element__VT4Ts inputs__form-element-text__13VAX"
>
  <label>
    label
  </label>
  <input
    max="10"
    min="0"
    onChange={[Function]}
    type="number"
    value="3"
  />
</div>
`;
